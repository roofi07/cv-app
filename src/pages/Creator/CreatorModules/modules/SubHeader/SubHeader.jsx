import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import PropTypes from 'prop-types'
import './SubHeader.scss'

export class SubHeader extends React.Component {
  static propTypes = {
    data: ImmutablePropTypes.map,
    onUpdate: PropTypes.func,
  }

  constructor() {
    super()

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(e) {
    const { onUpdate } = this.props
    let { data } = this.props
    const { value } = e.target

    data = data.set('value', value)
    onUpdate(data)
  }

  render() {
    const { data } = this.props
    return (
      <div className="subheader">
        <input onChange={this.handleChange} placeholder="Enter subheader..." value={data.get('value', '')} />
      </div>
    )
  }
}

export default SubHeader
