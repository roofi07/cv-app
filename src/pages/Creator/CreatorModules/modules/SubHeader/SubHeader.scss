.subheader {
  padding: 10px;
  padding-left: 204px;

  input {
    width: 100%;
    padding: 6px;
    font-size: 24px;
    border: 1px solid transparent;  

    &:focus {
      outline: 0
    }

    &:hover,
    &:focus {
      border-color: #dedede;        
    }
  }
}
