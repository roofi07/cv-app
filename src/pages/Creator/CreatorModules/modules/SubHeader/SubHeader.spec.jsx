import React from 'react'
import { shallow } from 'enzyme'
import Immutable from 'immutable'
import { SubHeader } from './SubHeader'

const data = {
  value: 'test',
}

const initPage = overrides => {
  const mockMethods = {
    onUpdate: jest.fn(),
  }
  const wrapper = shallow(<SubHeader
    data={Immutable.fromJS(data)}
    {...mockMethods}
    {...overrides}
  />)
  return { mockMethods, wrapper }
}

describe('SubHeader', () => {
  it('should render as expected', () => {
    const { wrapper } = initPage()
    expect(wrapper).toMatchSnapshot()
  })
})
