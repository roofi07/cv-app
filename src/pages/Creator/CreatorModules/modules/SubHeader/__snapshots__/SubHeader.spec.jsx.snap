// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader should render as expected 1`] = `
<div
  className="subheader"
>
  <input
    onChange={[Function]}
    placeholder="Enter subheader..."
    value="test"
  />
</div>
`;
